src/webportal/src/app/job/job-view/fabric/task-attempt/task-attempt-list.jsx

```diff
@@ -0,0 +1,674 @@
+          'com.microsoft.pai.runtimeplugin'
+        ]) {
+          if (pluginSetting.plugin === 'ssh') {
+            if (
+              'parameters' in pluginSetting &&
+              'userssh' in pluginSetting.parameters &&
+              !isEmpty(pluginSetting.parameters.userssh)
+            ) {
+              hasUserSsh = true;
+              break;
+            }
+          }
+        }
+      }
+      if (hasUserSsh) {
+        res.push(
+          'You can connect to this container by one of the following commands if SSH is set up properly: \n',
+        );
+        res.push(`1. Use your default SSH private key:\n`);
+        res.push(`ssh -p ${containerPorts.ssh} root@${containerIp}\n`);
+        res.push(`2. Use a pre-downloaded SSH private key:\n`);
+        res.push(
+          `On Windows:\nssh -p ${containerPorts.ssh} -i <your-private-key-file-path> root@${containerIp}\n`,
+        );
+        res.push(
+          `On Unix-like System:\nchmod 400 <your-private-key-file-path> && ssh -p ${containerPorts.ssh} -i <your-private-key-file-path> root@${containerIp}\n\n`,
+        );
+        res.push(
+          `If you are using a different username in your docker, please change "root" to your pre-defined username.`,
+        );
+      } else {
+        res.push('This job does not contain SSH info.');
+        res.push(
+          'If you want to use SSH, please enable it in the "Tools -> SSH" Section on the Job Submission Page.',
+        );
+      }
+      this.setState({
+        monacoProps: {
+          value: res.join('\n'),
+          options: {
+            wordWrap: 'off',
+            readOnly: true,
+          },
+        },
+        monacoTitle: `SSH to ${id}`,
+      });
+    }
+  }
+
+  getTaskPropertyFromColumnKey(item, key) {
+    if (key === 'exitType') {
+      return !isNil(item.containerExitSpec) &&
+        !isNil(item.containerExitSpec.type)
+        ? item.containerExitSpec.type
+        : null;
+    }
+    return item[key];
+  }
+
+  onRenderRow(props) {
+    return (
+      <DetailsRow
+        {...props}
+        styles={{
+          root: {
+            color: theme.palette.black,
+          },
+        }}
+      />
+    );
+  }
+
+  render() {
+    const { monacoTitle, monacoProps, monacoFooterButton, items } = this.state;
+    return (
+      <div>
+        <DetailsList
+          styles={{ root: { overflow: 'auto' } }}
+          columns={this.getColumns()}
+          disableSelectionZone
+          items={items}
+          layoutMode={DetailsListLayoutMode.justified}
+          selectionMode={SelectionMode.none}
+          onRenderRow={this.onRenderRow}
+        />
+        {/* Monaco Editor Panel */}
+        <MonacoPanel
+          isOpen={!isNil(monacoProps)}
+          onDismiss={this.onDismiss}
+          title={monacoTitle}
+          monacoProps={monacoProps}
+          footer={monacoFooterButton}
+        />
+      </div>
+    );
+  }
+
+  getColumns() {
+    const defaultColumns = [
```

Comment:
For the Task Attempt List table, sort by Attempt Index from large (new) to small (old)?

Reply:
done